packages/protocol/contracts/governance/Validators.sol

```diff
@@ -693,150 +504,110 @@ contract Validators is IValidators, Ownable, ReentrancyGuard, Initializable, Usi
   )
     external
     view
-    returns (string memory, string memory, string memory, address[] memory)
+    returns (string memory, string memory, address[] memory, uint256)
   {
     require(isValidatorGroup(account));
     ValidatorGroup storage group = groups[account];
-    return (group.identifier, group.name, group.url, group.members.getKeys());
+    return (group.name, group.url, group.members.getKeys(), group.commission.unwrap());
   }
 
   /**
-   * @notice Returns electable validator group addresses and their vote totals.
-   * @return Electable validator group addresses and their vote totals.
+   * @notice Returns the number of members in a validator group.
+   * @param account The address of the validator group.
+   * @return The number of members in a validator group.
    */
-  function getValidatorGroupVotes() external view returns (address[] memory, uint256[] memory) {
-    return votes.getElements();
+  function getGroupNumMembers(address account) public view returns (uint256) {
+    return groups[account].members.numElements;
   }
 
   /**
-   * @notice Returns the number of votes a particular validator group has received.
-   * @param group The account that registered the validator group.
-   * @return The number of votes a particular validator group has received.
+   * @notice Returns the top n group members for a particular group.
+   * @param account The address of the validator group.
+   * @param n The number of members to return.
+   * @return The top n group members for a particular group.
    */
-  function getVotesReceived(address group) external view returns (uint256) {
-    return votes.getValue(group);
+  function getTopValidatorsFromGroup(
+    address account,
+    uint256 n
+  )
+    external
+    view
+    returns (address[] memory)
+  {
+    address[] memory topAccounts = groups[account].members.headN(n);
+    address[] memory topValidators = new address[](n);
+    for (uint256 i = 0; i < n; i = i.add(1)) {
```

Comment:
safe to do `i++`. we know group.member.length < maxGroupSize

Reply:
Same question elsewhere